utilities/ReportingUtilities.groovy

```diff
@@ -0,0 +1,356 @@
+					// write impactedFiles per application to build workspace
+					// naming convention: externalImpacts_<collectionName>.log
+					String encodedFileName = URLEncoder.encode("externalImpacts_${entry.key}.log", "UTF-8")
+					String impactListFileLoc = "${props.buildOutDir}/${encodedFileName}"
+					if (props.verbose) println("*** Writing report of external impacts to file $impactListFileLoc")
+					File impactListFile = new File(impactListFileLoc)
+					String enc = props.logEncoding ?: 'IBM-1047'
+					impactListFile.withWriter(enc) { writer ->
+						externalImpactList.each { file ->
+							// if (props.verbose) println file
+							writer.write("$file\n")
+						}
+					}
+				}
+			}
+
+		}
+		else {
+			println("*! build property reportExternalImpactsAnalysisDepths has an invalid value : ${props.reportExternalImpactsAnaylsisDepths} , valid: simple | deep")
+		}
+
+	} catch (Exception e) {
+		println("*! (ReportingUtilities.reportExternalImpacts) Exception caught during reporting of external impacts. Build continues.")
+		println(e.getMessage())
+		println(e.printStackTrace())
+	}
+}
+
+/*
+ * Used to inspect dbb collections for potential impacts, sub-method to reportExternalImpacts
+ */
+
+def calculateLogicalImpactedFiles(List<String> fileList, Set<String> changedFiles, Map<String,HashSet> collectionImpactsSetMap, String indentationMsg, String analysisMode) {
+	MetadataStore metadataStore = MetadataStoreFactory.getMetadataStore()
+
+	// local matchers to inspect files and collections
+	List<Pattern> collectionMatcherPatterns = createMatcherPatterns(props.reportExternalImpactsCollectionPatterns)
+
+	// local
+	List<LogicalDependency> logicalDependencies = new ArrayList()
+	
+	// will be returned
+	Set<String> impactedFiles = new HashSet<String>()
+
+	// creating a list logical dependencies
+	fileList.each{ file ->
+		// go after all the files passed in; assess the identified impacted files to skip analysis for files from an impactSet which are on the changed files
+		if(analysisMode.equals('buildSet') || (analysisMode.equals('impactSet') && !changedFiles.contains(file))){
+			String memberName = CopyToPDS.createMemberName(file)
+			def ldepFile = new LogicalDependency(memberName, null, null);
+			logicalDependencies.add(ldepFile)
+		}else {
+			// debug-output
+			// println("$indentationMsg!* Skipped redundant analysis. $file was already or will be procceed soon.")
+		}
+	}
+
+	if(logicalDependencies.size != 0) {
+
+		// iterate over collections
+		metadataStore.getCollections().each{ collection ->
+			String cName = collection.getName()
+			if(matchesPattern(cName,collectionMatcherPatterns)){ // find matching collection names
+
+				def Set<String> externalImpactList = collectionImpactsSetMap.get(cName) ?: new HashSet<String>()
+				// query dbb metadatastore for files with all logicalDependencies
+				def logicalImpactedFilesCollections = metadataStore.getImpactedFiles([cName], logicalDependencies);
```

Comment:
Rather than querying for each individual collection name, pass the list of collections for which the filter applies.